### PR TITLE
feat(epd): update product image field

### PR DESCRIPTION
**Pull request by irynadosiak-c-change-lab**

### CODE REVIEW

**Comment by corvis** [comment]
This will kill validation. Unfortunately pydantic v1 doesn't have a good solution for type checking of this kind of fields. All you can do is either wrap values in AnyUrl before setting or type ignore, or cast.

File: src/openepd/model/epd.py [hunk 1]
```diff
@@ -271,3 +271,10 @@ def validate_compliance(cls, v):
         if v is None:
             return []
         return v
+
+    @pyd.validator("product_image", always=True, pre=True)
+    def validate_product_image(cls, v):
```
Comment (alexlitvinenko) [inline src/openepd/model/epd.py:276]: Add type hints

File: src/openepd/model/epd.py [hunk 2]
```diff
@@ -271,3 +271,10 @@ def validate_compliance(cls, v):
...
+        """Handle correctly validation for product image field.."""
```
Comment (alexlitvinenko) [inline src/openepd/model/epd.py:277]: Redundant dot

File: src/openepd/model/epd.py [hunk 3]
```diff
@@ -271,3 +271,10 @@ def validate_compliance(cls, v):
...
+        if v is not None and isinstance(v, str) and v.startswith("file://"):
```
Comment (alexlitvinenko) [inline src/openepd/model/epd.py:278]: Remove `v is not None` because you are doing `is_instance` further.

**Review by alexlitvinenko — CHANGES_REQUESTED** [review]

File: src/openepd/model/epd.py (hunk 1 above)
Comment (irynadosiak-c-change-lab) [reply src/openepd/model/epd.py]: added

File: src/openepd/model/epd.py (hunk 3 above)
Comment (irynadosiak-c-change-lab) [reply src/openepd/model/epd.py]: removed

**Comment by alexlitvinenko** [comment]
> Or do you think the current implementation is not sufficient?

It is ok, I didn't get what was the problem for the first time.

**Comment by corvis** [comment]
Maybe just `product_image: pyd.AnyHttpUrl | pyd.FileUrl`?

**Comment by irynadosiak-c-change-lab** [comment]
> Maybe just `product_image: pyd.AnyHttpUrl | pyd.FileUrl`?

Oh, you're a genius)
Thank you, updated

**Review by corvis — APPROVED** [review]

**Pull request merged**